src/main/java/core/basesyntax/strategy/impl/PurchaseService.java

```diff
@@ -5,7 +5,7 @@
 import core.basesyntax.model.FruitTransaction;
 
 public class PurchaseService implements OperationHandler {
-    private FruitDao fruitDao = new FruitDaoImpl();
+    private final FruitDao fruitDao = new FruitDaoImpl();

```

Comment:
It is a good practice to inject service objects using a constructor. Any class must not pe responsible for creating its dependencies